contracts/Repayer.sol

```diff
@@ -0,0 +1,259 @@
+        CCTP_TOKEN_MESSENGER = ICCTPTokenMessenger(cctpTokenMessenger);
+        CCTP_MESSAGE_TRANSMITTER = ICCTPMessageTransmitter(cctpMessageTransmitter);
+    }
+
+    function initialize(
+        address admin,
+        address repayer,
+        address[] memory pools,
+        Domain[] memory domains,
+        Provider[] memory providers,
+        bool[] memory poolSupportsAllTokens
+    ) external initializer() {
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+        _grantRole(REPAYER_ROLE, repayer);
+        _setRoute(pools, domains, providers, poolSupportsAllTokens, true);
+    }
+
+    function setRoute(
+        address[] calldata pools,
+        Domain[] calldata domains,
+        Provider[] calldata providers,
+        bool[] memory poolSupportsAllTokens,
+        bool isAllowed
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        _setRoute(pools, domains, providers, poolSupportsAllTokens, isAllowed);
+    }
+
+    function initiateRepay(
+        IERC20 token,
+        uint256 amount,
+        address destinationPool,
+        Domain destinationDomain,
+        Provider provider,
+        bytes calldata /*extraData*/
+    ) external override onlyRole(REPAYER_ROLE) {
+        require(amount > 0, ZeroAmount());
+        require(token.balanceOf(address(this)) >= amount, InsufficientBalance());
+        require(isRouteAllowed(destinationPool, destinationDomain, provider), RouteDenied());
+
+        emit InitiateRepay(token, amount, destinationPool, destinationDomain, provider);
+
+        RepayerStorage storage $ = _getStorage();
+
+        if (provider == Provider.LOCAL) {
+            // This should always pass because isRouteAllowed check will fail earlier.
+            // It is put here for explicitness.
+            require(destinationDomain == DOMAIN, UnsupportedDomain());
+
+            if (!$.poolSupportsAllTokens[destinationPool]) {
+                require(token == ASSETS, InvalidToken());
+            }
+            // For local we proceed to the process right away.
+            _processRepayLOCAL(token, amount, destinationPool);
+        } else
+        if (provider == Provider.CCTP) {
+            require(token == ASSETS, InvalidToken());
+            initiateTransferCCTP(CCTP_TOKEN_MESSENGER, ASSETS, amount, destinationPool, destinationDomain);
+        } else {
+            // Unreachable atm, but could become so when more providers are added to enum.
+            revert UnsupportedProvider();
+        }
+    }
+
+    function processRepay(
+        address destinationPool,
+        Provider provider,
+        bytes calldata extraData
+    ) external override onlyRole(REPAYER_ROLE) {
+        require(isRouteAllowed(destinationPool, DOMAIN, Provider.LOCAL), RouteDenied());
+        uint256 amount = 0;
+        if (provider == Provider.CCTP) {
+            amount = processTransferCCTP(CCTP_MESSAGE_TRANSMITTER, ASSETS, destinationPool, extraData);
+            require(amount > 0, ZeroAmount());
```

Comment:
This is unreachable cause processTransferCCTP validates that the amount is positive.